modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -77,8 +77,9 @@
             }
             XContentBuilder builder = XContentFactory.contentBuilder(parser.contentType());
             builder.map(source);
-            parser = parser.contentType().xContent().createParser(builder.bytes());
```

Comment:
I found it strange that we re-purpose the parser instance here. I changed it to use a new local reference.